backend/src/Backend/Cache.hs

```diff
@@ -347,13 +345,14 @@ cacheBuild se cache cj = do
           liftIO $ putStrLn "Error uploading nix-cache-info\n"
           throwError (ExitFailure status)
 
-    results <- liftIO $ forM toCache $ \sp -> do
-      res <- cacheStorePath se e logProcMsg nixDbConn cache (StorePath $ T.unpack sp)
-      case res of
-        Left er -> do
-          liftIO $ logStr CiMsg $ T.pack $ printf "Error while caching %s:\n%s" sp (show er)
-        Right _ -> return ()
-      return res
+    results <- liftIO $ bracket (open nixSqliteDb) close $ \nixDbConn -> 
```

Comment:
`withConnection` (from `Database.SQLite.Simple`, which is also where `open` comes from) is defined as
```
\conn -> bracket (open connection) close
```

You should probably use that here instead of writing out the `bracket`